src/syscheckd/src/db/testtool/action.h

```diff
@@ -0,0 +1,401 @@
+ */
+
+#ifndef _ACTION_H
+#define _ACTION_H
+#include <json.hpp>
+#include <mutex>
+#include <iostream>
+#include <sstream>
+#include <fstream>
+#include "commonDefs.h"
+#include "dbsync.hpp"
+#include "testContext.h"
+#include "db.hpp"
+
+
+struct IAction
+{
+    virtual void execute(std::unique_ptr<TestContext>& ctx, const nlohmann::json& value) = 0;
+
+    virtual ~IAction() = default;
+};
+
+struct RemoveFileAction final : public IAction
+{
+    void execute(std::unique_ptr<TestContext>& ctx, const nlohmann::json& value) override
+    {
+        auto retVal = false;
+        try
+        {
+            DB::instance().removeFile(value.at("file_path").get_ref<const std::string&>());
+            retVal = true;
+        }
+        catch (const std::exception &e)
+        {
+            std::cout << "Error removing file: "
+                << value.at("file_path").get_ref<const std::string&>() << std::endl;
+        }
+
+        std::stringstream oFileName;
+        oFileName << "action_" << ctx->currentId << ".json";
+        const auto outputFileName{ ctx->outputPath + "/" + oFileName.str() };
+
+        std::ofstream outputFile{ outputFileName };
+        const nlohmann::json jsonResult = {
+                { "result", retVal },
+                { "action", "RemoveFile" }
+            };
+        outputFile << jsonResult.dump() << std::endl;
+    }
+};
+
+struct GetFileAction final : public IAction
+{
+    void execute(std::unique_ptr<TestContext>& ctx, const nlohmann::json& value) override
+    {
+        auto retVal = false;
+        nlohmann::json jsonReturn;
+        try
+        {
+            DB::instance().getFile(value.at("file_path").get_ref<const std::string&>(),
+                [&jsonReturn](const nlohmann::json& file) {
+                    jsonReturn = file;
+                });
+            retVal = true;
+        }
+        catch (const std::exception &e)
+        {
+            std::cout << "Error getting file: "
+                << value.at("file_path").get_ref<const std::string&>() << std::endl;
+        }
+        std::stringstream oFileName;
+        oFileName << "action_" << ctx->currentId << ".json";
+        const auto outputFileName{ ctx->outputPath + "/" + oFileName.str() };
+
+        std::ofstream outputFile{ outputFileName };
+        const nlohmann::json jsonResult = {
+                { "result", retVal },
+                { "value", jsonReturn },
+                { "action", "GetFile" }
+            };
+        outputFile << jsonResult.dump() << std::endl;
+    }
+};
+
+struct CountFilesAction final : public IAction
+{
+    void execute(std::unique_ptr<TestContext>& ctx, const nlohmann::json& value) override
+    {
+        auto retVal = false;
+        int count = 0;
+        try
+        {
+            const auto filterType
+            {
+                static_cast<COUNT_SELECT_TYPE>(value.at("filter_type").get<int32_t>())
+            };
+            count = DB::instance().countFiles(filterType);
```

Comment:
This has changed in 9103, and the way to call it is : `count = DB::instance().countEntries(FIMDB_FILE_TABLE_NAME, COUNT_SELECT_TYPE::COUNT_ALL);`
After this change the filterType variable will be useless